### PR TITLE
- fixed --root option (bsc#1016633)

### PR DESCRIPTION
**Pull request by aschnell**

### CODE REVIEW

File: client/proxy-lib.h [hunk 1]
```diff
@@ -86,8 +86,8 @@ class ProxySnapperLib : public ProxySnapper
 
 public:
 
-    ProxySnapperLib(const string& config_name)
-	: snapper(new Snapper(config_name, "/")), proxy_snapshots(this)
+    ProxySnapperLib(const string& config_name, const string& target_root)
```
Comment (jreidinger) [inline client/proxy-lib.h]: this is just some internal class? I miss any documentation here. At least link to what it proxy.

**Comment by jreidinger** [comment]
in general I miss testcase as it is very often used option in yast upgrade, I will welcome to have at least small trivial test.

**Comment by aschnell** [comment]
Yes, integration tests are missing as I mentioned in the call. Many are needed: 19 commands, --no-dbus, --root -> that makes already 57 tests. Additional many commands have several major modes, e.g. two for rollback, 2 for cleanup, 4 for create, ... And I'm not even talking about tests for wrong usage.

But so far always manual testing was considered enough by the PO.

File: client/proxy-lib.h (hunk 1 above)
Comment (aschnell) [inline client/proxy-lib.h]: Yes, it is a internal class. The concept is explained in proxy.h.

**Comment by jreidinger** [comment]
ok, so LGTM

**Pull request merged**